feat(rdb): seamless upgrade v (#2435)
scaleway/docs-content: managed-databases/postgresql-and-mysql/how-to/upgrade-version.mdx

  - postgresql-and-mysql
---

> If your database engine is outdated, you can upgrade the version the latest one anytime via the Scaleway console.

Reviewer: ```suggestion
If your database engine is outdated, you can upgrade the version to the latest one anytime via the Scaleway console.
```

> We especially recommend that you upgrade your engine version if your Database Instance uses an End of Life engine version that is no longer maintained.

Reviewer 2: ```suggestion
<Message tyepe="note>
  We recommend that you upgrade your engine version if your Database Instance uses an End of Life engine version that is no longer maintained.
</Message>
```

  - postgresql-and-mysql
---

If your database engine is outdated, you can upgrade the version to the latest one anytime via the Scaleway console.

<Message tyepe="note>

>   We recommend that you upgrade your engine version if your Database Instance uses an End of Life engine version that is no longer maintained.

Reviewer 3: ```suggestion
  You must upgrade your engine version if your Database Instance uses an End of Life engine version that is no longer maintained.
```

</Message>

  <Message type="important">
    The version upgrade is only available for:

>       - Database Instances using outdated engine versions.

Reviewer 3: the major version upgrade is available enven if the engine version is not end of life. I don't know if `outdated` is the right word for it 😅 

i.e you can upgrade a PostgreSQL 13 to PostgreSQL 14 or 15 even if PostgreSQL 13 EOL is scheduled to `November 13, 2025`

      - Database Instances using Block volumes. If your Database Instance uses a local volume, you must first [change the volume type](/managed-databases/postgresql-and-mysql/how-to/change-volume-type) to a Block volume and try again.
  </Message>

<Macro id="iam-requirements" />

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)

>   - You have a [PostgreSQL or MySQL Database Instance](/managed-databases/postgresql-and-mysql/how-to/create-a-database) running on an outdated engine version

Reviewer 3: We don't have yet major upgrade for MySQL

3. Select the upgrade method. Two methods are available:
    - **Upgrade only**: This method has no impact on your original Database Instance. Your Database Instance will remain available with its original endpoint and will continue to be billed.
    - **Upgrade and switch incoming traffic**: With this method your endpoint is migrated to the new Database Instance automatically. This option will create a clone of your Database Instance, and will automatically migrate the endpoint to the new Instance. The original Database Instance remains available, and will continue to be billed. The endpoint will be deleted from the original Database Instance.

> 

Reviewer 3: We must add an info here as it is not indicated. 

When the client choose to Upgrade its instance, there is no synchronisation between the source instance and the new upgraded instance. If the client to not want to loose data, the client application have to stop writting operation while the upgrade procedure is in progress. Otherwise data will still be stored in the old database and the new upgraded one won't have those new data

3. Select the upgrade method. Two methods are available:
    - **Upgrade only**: This method has no impact on your original Database Instance. Your Database Instance will remain available with its original endpoint and will continue to be billed.
    - **Upgrade and switch incoming traffic**: With this method your endpoint is migrated to the new Database Instance automatically. This option will create a clone of your Database Instance, and will automatically migrate the endpoint to the new Instance. The original Database Instance remains available, and will continue to be billed. The endpoint will be deleted from the original Database Instance.

    <Message type="important">
      When you upgrade to a new version, there is no synchronisation between the source and target Database Instances. To avoid data loss, we recommend you stop any write operations running on your applications during the upgrade. If you let them run during the process, the data will be stored only in the source Database Instance.
    </Message>

    <Message type="tip">

>       Follow the [migrating endpoints via the CLI](/managed-databases/postgresql-and-mysql/api-cli/migrating-endpoints) procedure to quickly migrate your endpoints. You can use this procedure to revert the migration anytime.

Reviewer 3: We can add a note there. Reverting the endpoint won't have effect on the date stored on the database. Meaning of some entries were added in the upgraded database, they won't be added back on the old version if the client choose to migrate back its endpoints
